silkworm/db/state/storage_codecs.hpp

```diff
@@ -34,26 +34,69 @@ struct Bytes32KVDBCodec : public datastore::kvdb::Codec {
         std::memcpy(value.bytes, slice.data(), sizeof(value.bytes));
     }
 };
-
 static_assert(datastore::kvdb::EncoderConcept<Bytes32KVDBCodec>);
 static_assert(datastore::kvdb::DecoderConcept<Bytes32KVDBCodec>);
 
+struct PackedBytes32KVDBCodec : public datastore::kvdb::Codec {
+    evmc::bytes32 value;
+    ~PackedBytes32KVDBCodec() override = default;
+
+    datastore::kvdb::Slice encode() override {
+        // find first non-zero byte
+        u_int64_t offset = 0;
+        while (value.bytes[offset] == 0 && offset < sizeof(value.bytes)) {
+            offset++;
+        }
+        return {value.bytes + offset, sizeof(value.bytes) - offset};
+    }
+
+    void decode(datastore::kvdb::Slice slice) override {
+        SILKWORM_ASSERT(slice.size() <= sizeof(value.bytes));
+        u_int64_t offset = sizeof(value.bytes) - slice.size();
+        std::memset(value.bytes, 0, offset);
+        std::memcpy(value.bytes + offset, slice.data(), slice.size());
```

Comment:
I suggest to use `to_bytes32` from `core/types/evmc_bytes32.hpp` to avoid code duplication